app/src/main/java/com/keylesspalace/tusky/pager/MainPagerAdapter.kt

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2017 Andrew Dawson
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.pager
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+import androidx.viewpager.widget.PagerAdapter
+import com.keylesspalace.tusky.TabData
+
+class MainPagerAdapter(val tabs: List<TabData>, manager: FragmentManager) : FragmentPagerAdapter(manager) {
+
+    override fun getItem(position: Int): Fragment {
+        return tabs[position].fragment()
+    }
+
+    override fun getCount(): Int {
+        return tabs.size
+    }
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return null
+    }
+
+    override fun getItemId(position: Int): Long {
+        return tabs[position].id.hashCode().toLong()
+    }
+
+    override fun getItemPosition(item: Any): Int {
+        return PagerAdapter.POSITION_NONE
```

Comment:
without this the adapter does not create new fragments when they change

Reply:
Can we force it somehow?..

Reply:
apparently this is how you force it

Reply:
This way they're created each time you scroll left-right or?..

Reply:
No I don't think so, but I will double check